.evergreen/buildvariants-and-tasks.yml

```diff
@@ -71,7 +71,7 @@ buildvariants:
     expansions:
       silk_asset_group: compass-macos-arm
     display_name: Package MacOS Arm64
-    run_on: macos-1100-arm64
+    run_on: macos-14-arm64
```

Comment:
Just for context, we've been building on a fairly low macOS version because we've felt that it gives us a better sense of guarantee that Compass will actually run on the macOS versions we claim to support. Currently, in our docs, that's 10.12 – which is definitely wrong at this point, Electron 27 raised the minimum to 10.15, so I'd recommend closing this ticket with "Documentation Changes Needed".

Now, I've chatted with some people in devprod before, but haven't really found anyone who is intimately familiar with what macOS's actual forward/backward compatibility story looks like in practice – i.e. answers to questions like "if we package on macOS 14, can we still relatively safely claim that it will run on 10.15 (or on 11, etc.)? Does it even really matter?".

All of which is to say: I think this change is fine, but in the long run we'll probably want to figure out what the actual story here is.